src/main/scala/firrtl/passes/ReplSeqMem.scala

```diff
@@ -0,0 +1,142 @@
+    outputBuffer.append(conf)
+  }
+  def serialize = {
+    val outputFile = new java.io.PrintWriter(filename)
+    outputFile.write(outputBuffer.toString)
+    outputFile.close()
+  }
+}
+
+case class ReplSeqMemAnnotation(t: String, tID: TransID)
+    extends Annotation with Loose with Unstable {
+
+  val usage = """
+[Optional] ReplSeqMem
+  Pass to replace sequential memories with blackboxes + configuration file
+
+Usage: 
+  --replSeqMem -c:<circuit>:-i:<filename>:-o:<filename>
+  *** Note: sub-arguments to --replSeqMem should be delimited by : and not white space!
+
+Required Arguments:
+  -o<filename>         Specify the output configuration file
+  -c<compiler>         Specify the target circuit
+
+Optional Arguments:
+  -i<filename>         Specify the input configuration file (for additional optimizations)
+"""    
+
+  val passOptions = PassConfigUtil.getPassOptions(t,usage)
+  val outputConfig = passOptions.getOrElse(
+    OutputConfigFileName, 
+    Error("No output config file provided for ReplSeqMem!" + usage)
+  )
+  val passCircuit = passOptions.getOrElse(
+    PassCircuitName, 
+    Error("No circuit name specified for ReplSeqMem!" + usage)
+  )
+  val target = CircuitName(passCircuit)
+  def duplicate(n: Named) = this.copy(t=t.replace("-c:"+passCircuit,"-c:"+n.name))
+  
+}
+
+class ReplSeqMem(transID: TransID) extends Transform with LazyLogging {
+  def execute(circuit:Circuit, map: AnnotationMap) = 
+    map get transID match {
+      case Some(p) => p get CircuitName(circuit.main) match {
+        case Some(ReplSeqMemAnnotation(t, _)) => {
+
+          val inputFileName = PassConfigUtil.getPassOptions(t).getOrElse(InputConfigFileName,"")
+          val inConfigFile = {
+            if (inputFileName.isEmpty) None 
+            else if (new java.io.File(inputFileName).exists) Some(new YamlFileReader(inputFileName))
+            else Error("Input configuration file does not exist!")
+          }
+
+          val outConfigFile = new ConfWriter(PassConfigUtil.getPassOptions(t).get(OutputConfigFileName).get)
+          TransformResult(
+            (
+              Seq(
+                Legalize,
+                AnnotateMemMacros,
+                UpdateDuplicateMemMacros,
+                new AnnotateValidMemConfigs(inConfigFile),
+                new ReplaceMemMacros(outConfigFile),
+                RemoveEmpty,
+                CheckInitialization,
+                ResolveKinds,                                       // Must be run for the transform to work!
+                InferTypes,
+                ResolveGenders
+              ) foldLeft circuit
+            ){ 
+              (c, pass) =>
+                val x = Utils.time(pass.name)(pass run c)
+                logger debug x.serialize
+                x
+            }, 
+            None, 
+            Some(map)
+          )
+        }  
+        case _ => TransformResult(circuit, None, Some(map))
```

Comment:
Error instead of doing nothing? If you get an annotation with the correct TransID but it isn't the annotation you were expecting, I believe this should be an error.